### PR TITLE
placement: fix nil error during FitRegion cache

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Fix nil error during FitRegion Cache. Previously, `SetCache` is called by `rule_checker` and `InvalidCache` is called by `Fitregion`. This made `Set` and `Invalid` would run in different goroutine and which may caused nil error in following situation.

```sh
     goroutine1  goroutine2 
1.   CheckCache
2.               Invalid Cache
3.   GetCache
```

### What is changed and how it works?

Let `rule_checker` invalid cache so that `set` and `invalid` will run in one goroutine. We also merge `check` and `get` in order to guarantee by the same mutex.

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by disksing** [comment]
why delete `TestFitRegionCache`?

**Comment by Yisaer** [comment]
> why delete `TestFitRegionCache`?

As the `invalidCache` in moved into `checker`, the related testcases are moved into `TestRuleCache`.

File: server/schedule/checker/rule_checker.go [hunk 1]
```diff
@@ -65,17 +65,23 @@ func (c *RuleChecker) Check(region *core.RegionInfo) *operator.Operator {
 }
 
 // CheckWithFit checkWithFit is similar with Checker with placement.RegionFit
```
Comment (JmPotato) [inline server/schedule/checker/rule_checker.go:67]:
```suggestion
// CheckWithFit is similar with Checker with placement.RegionFit
```

File: server/schedule/placement/rule_manager.go [hunk 2]
```diff
@@ -328,6 +324,11 @@ func (m *RuleManager) SetRegionFitCache(region *core.RegionInfo, fit *RegionFit)
 	m.cache.SetCache(region, fit)
 }
 
+// InvalidCache invalid cache
```
Comment (JmPotato) [inline server/schedule/placement/rule_manager.go:327]:
```suggestion
// InvalidCache invalids the cache.
```

**Review by JmPotato — APPROVED** [review]
The rest LGTM.

**Review by disksing — APPROVED** [review]

**Pull request merged**